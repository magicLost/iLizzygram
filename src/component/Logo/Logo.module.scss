.image {
  width: 80px;
  height: auto;
}

.iconButton {
  padding: 5px;
}
