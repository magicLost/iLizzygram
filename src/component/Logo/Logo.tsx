import React from "react";
import IconButton from "@material-ui/core/IconButton";
//import classes from './Logo.module.scss';
import { makeStyles } from "@material-ui/core/styles";
import { Link } from "gatsby";

interface LogoProps {}

const useStyles = makeStyles({
  iconButton: {
    padding: "20px",
  },
});

const Logo = ({}: LogoProps) => {
  const classes = useStyles();
  return (
    <IconButton
      aria-label="Home page"
      className={classes.iconButton}
      component={Link}
      to="/"
    >
      <img src="logo.svg" width={"80px"} alt="Lizzygram logo" />
    </IconButton>
  );
};

export default Logo;
