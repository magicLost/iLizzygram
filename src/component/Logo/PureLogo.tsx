import React from "react";
import IconButton from "@material-ui/core/IconButton";
//import classes from './Logo.module.scss';
import { makeStyles } from "@material-ui/core/styles";
//import { Link } from "gatsby";

//interface LogoProps {}

const useStyles = makeStyles({
  iconButton: {
    padding: "5px",
  },
});

export const PureLogo = ({ ILink }) => {
  const classes = useStyles();
  return (
    <IconButton
      aria-label="Home page"
      className={classes.iconButton}
      component={ILink}
      to="/"
    >
      <img src="logo.svg" width={"80px"} alt="Lizzygram logo" />
    </IconButton>
  );
};

export default PureLogo;
