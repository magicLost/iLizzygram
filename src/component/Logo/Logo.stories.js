import React from "react";
import { action } from "@storybook/addon-actions";
import Logo from "./Logo";
//import NextRouter from "next/router";

export default {
  component: Logo,
  title: "Logo",
  decorators: [],
  //decorators: [withKnobs],
  // Our exports that end in "Data" are not stories.
  excludeStories: /.*Data$/,
};

export const Default = () => {
  return (
    <div style={{ margin: "auto", backgroundColor: "white" }}>
      <Logo />
    </div>
  );
};
