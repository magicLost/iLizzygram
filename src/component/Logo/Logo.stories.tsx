import React from "react";
//import { action } from "@storybook/addon-actions";
import Logo from ".";
import { withNextRouter } from "storybook-addon-next-router";

export default {
  component: Logo,
  title: "Components/Logo",
  decorators: [
    withNextRouter,
    story => (
      <div
        style={{
          width: "100px",
          margin: "auto",
          paddingTop: "30px",
          backgroundColor: "white",
        }}
      >
        {story()}
      </div>
    ),
  ],
  //decorators: [withKnobs],
  // Our exports that end in "Data" are not stories.
  excludeStories: /.*Data$/,
};

/* const Link = React.forwardRef((props, ref) => {
  return (
    <a href={props.to} ref={ref}>
      {props.children}
    </a>
  );
}); */

const Template = args => <Logo {...args} />;

export const Default = Template.bind({});
