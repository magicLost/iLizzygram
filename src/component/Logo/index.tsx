import React from "react";
import IconButton from "@material-ui/core/IconButton";
import classes from "./Logo.module.scss";
//import { makeStyles } from "@material-ui/core/styles";
//import { Link } from "gatsby";
import ButtonLink from "../UI/ButtonLink";
//import logo from "./../../../static/logo.svg";

//interface LogoProps {}

/* const useStyles = makeStyles({
  iconButton: {
    padding: "5px",
  },
}); */

const Logo = () => {
  //const classes = useStyles();

  console.log("[RENDER LOGO]");
  return (
    <IconButton
      aria-label="Home page"
      className={classes.iconButton}
      component={ButtonLink}
      href="/"
      hrefAs="/"
    >
      <img
        width="80px"
        src={"logo.svg"}
        className={classes.image}
        alt="Lizzygram лого"
      />
    </IconButton>
  );
};

export default Logo;

/* import React from "react";
import PureLogo from "./PureLogo";
import { Link } from "gatsby";



const Logo = () => {
  return <PureLogo ILink={Link} />;
};

export default Logo;
 */
