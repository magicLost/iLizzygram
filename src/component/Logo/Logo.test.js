import React from "react";
import {
  render,
  fireEvent,
  cleanup,
  waitForElement,
} from "@testing-library/react";
import { configure } from "@testing-library/dom";
import "@testing-library/jest-dom/extend-expect";

import Logo from ".";
import classes from "./Logo.module.scss";

jest.mock("gatsby", () => {
  const RReact = require("react");
  return {
    __esModule: true,
    Link: RReact.forwardRef((props, ref) => (
      <a href={props.to} ref={ref}>
        {props.children}
      </a>
    )),
    /* Link: props => {
      return <a href={props.to}>{props.children}</a>;
    }, */
  };
});

describe("Logo", () => {
  let _render = null;

  describe("Render and props test", () => {
    beforeEach(() => {
      _render = render(<Logo />);
    });

    afterEach(cleanup);

    describe("Snapshots", () => {
      test("matches snapshot", () => {
        const { baseElement } = _render;
        expect(baseElement).toMatchSnapshot();
      });
    });
  });
});
