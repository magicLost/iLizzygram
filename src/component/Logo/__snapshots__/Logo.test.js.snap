// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo Render and props test Snapshots matches snapshot 1`] = `
<body>
  <div>
    <a
      href="/"
    >
      <span
        class="MuiIconButton-label"
      >
        <img
          alt="Lizzygram logo"
          src="logo.svg"
          width="80px"
        />
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </a>
  </div>
</body>
`;
